docs/pages/versions/unversioned/sdk/firebase-recaptcha.md

```diff
@@ -83,29 +85,32 @@ import { getAuth, PhoneAuthProvider } from 'firebase/auth';
   // ignore app already initialized error in snack
 }*/
 
+// Firebase references
+const app = getApp();
+const auth = getAuth();
+
+// Double-check that we can run the example
+if (!app?.options || Platform.OS === 'web') {
+  throw new Error('This example only works on Android or iOS, and requires a valid Firebase config.');
+}
+
 export default function App() {
+  // Ref or state management hooks
   const recaptchaVerifier = React.useRef(null);
   const [phoneNumber, setPhoneNumber] = React.useState();
   const [verificationId, setVerificationId] = React.useState();
   const [verificationCode, setVerificationCode] = React.useState();
-  const app = getApp();
+
   const firebaseConfig = app ? app.options : undefined;
-  const [message, showMessage] = React.useState(
-    !firebaseConfig || Platform.OS === 'web'
-      ? {
-          text:
-            'To get started, provide a valid firebase config in App.js and open this snack on an iOS or Android device.',
-        }
-      : undefined
-  );
+  const [message, showMessage] = React.useState();
   const attemptInvisibleVerification = false;
 
   return (
     <View style={{ padding: 20, marginTop: 50 }}>
       <FirebaseRecaptchaVerifierModal
         ref={recaptchaVerifier}
-        firebaseConfig={firebaseConfig}
-        attemptInvisibleVerification={attemptInvisibleVerification}
+        firebaseConfig={app.options}
```

Comment:
Is there a reason why you did not have used the `firebaseConfig` variable in here?

Reply:
Usually, you don't want to reuse the firebase config directly in your RN app. E.g. if you need to test, you can just mock the app and you will have a handle on this as well. Or if you have a provider, you probably don't want to leak the config to all your components using this captcha.

It's one of many ways to provide this config, I think this is the most versatile one. Where you can just initialize that app with whatever you need, and only that will be used. I can revert it though, don't have a heavy opinion on this